src/Packetery/Module/Order/PacketAutoSubmitter.php

```diff
@@ -125,7 +126,7 @@ public function handleEvent( string $event, int $orderId, ?bool $triggeredByFron
 		assert( null !== $wcOrder, 'WC order has to be present' );
 
 		$paymentGateway = wc_get_payment_gateway_by_order( $wcOrder );
-		if ( false === $this->optionsPage->hasPaymentGateway( $paymentGateway ) ) {
+		if ( ! $paymentGateway instanceof WC_Payment_Gateway || false === $this->optionsPage->hasPaymentGateway( $paymentGateway ) ) {
```

Comment:
Metoda hasPaymentGateway ve skutečnosti kontroluje, jestli patří poskytnutá platební metoda mezi dostupné. Má tedy zavádějící název. Vzhledem k tomu, že se používá jen v této třídě a je to oneliner, přesunul bych kód přímo sem a zrušil bych ji.

Reply:
To není pravda, můžeš si všimnout, že v rámci PES-1419 závislost na OptionsPage mizí a je to komplet statické. Metoda kontroluje něco obecnějšího, jak píšu výše.